Add correlating logs blog (#1769)
opensearch-project/project-website: _posts/2023-07-13-correlating-security-events.md

Before we dive deeper on Security Analytics and correlation engine, it’s useful to know a couple of important terms related to threat detection.

### Detectors

**Detectors** are core components that are configured to identify a range of cybersecurity threats corresponding to an ever-growing knowledge base of adversary tactics and techniques maintained by the [MITRE ATT&CK](https://attack.mitre.org/) organization. Detectors use log data to evaluate events occurring in the system. They then apply a set of security rules specified for the detector and determine findings from these events. 

### Findings

> **Findings** are generated every time a detector matches a rule with a log event. Findings do not necessarily point to imminent threats within the system, but they always isolate an event of interest. Because they represent the result of a specific definition for a detector, findings include a unique combination of select rules, a log type, and a rule severity. As such, you can search for specific findings in the Findings window, and you can filter findings in the list based on severity and log type.

Author: ```suggestion
**Findings** are generated every time a detector matches a rule with a log event. Findings do not necessarily point to imminent threats within the system, but they always isolate an event of interest. Because they represent the result of a specific definition for a detector, findings include a unique combination of selected rules, a log type, and a rule severity. As such, you can search for specific findings in the Findings window, and you can filter findings in the list based on severity and log type.
```

The following sample customer infrastructure diagram further illustrates the need for a correlation framework. This customer deploys a Django REST application hosted on a EC2 Windows instance. The REST APIs for the application use Active Directory as the identity provider and Amazon S3 to store and query files. The EC2 Windows instance stores incoming network traffic in VPC Flow Logs.

<img src="/assets/media/blog-images/2023-07-13-correlating-security-events/customer application.png" alt="Sample customer infrastructure"/>{: .img-fluid }

For Security Analytics to monitor and detects threats for this infrastructure, we need to the define a detector for each component in the infrastructure. The following diagram shows the infrastructure with a detector for each component, such as a network detector for VPC Flow logs and a Windows detector for the EC2 Windows instance.

<img src="/assets/media/blog-images/2023-07-13-correlating-security-events/detector-infrastructure.png" alt="Detector infrastructure inside of customer's application"/>{: .img-fluid }

> With the detectors set, we can stimulate a security attack on this example infrastructure. The attacker uses `sbcd90` user to call a REST API named `POST /customer_records.txt` which tries to replicate a sensitive file named` customer_records.txt` from S3.

Reviewer: ```suggestion
With the detectors set, we can simulate a security attack on this example infrastructure. The attacker uses the `sbcd90` user to call a REST API named `POST /customer_records.txt`, which tries to replicate a sensitive file named `customer_records.txt` from Amazon S3.
```

> When the attack occurs, each detector generates a finding for its corresponding infrastructure component. For example, the AD/LDAP detector generates a finding that indicates `Invalid Username/Password, ResultType: 50126`. This finding belongs to a that individual detectors respective log type. 

Reviewer: ```suggestion
When the attack occurs, each detector generates a finding for its corresponding infrastructure component. For example, the AD/LDAP detector generates a finding that indicates `Invalid Username/Password, ResultType: 50126`. This finding belongs to the individual detector's log type. 
```

> But how would you know that the AD/LDAP detector finding `Invalid Username/Password, ResultType: 50126` is related to a chain of security events occurring in the same time window on the infrastructure, such a `403 Forbidden error` finding from an Application logs detector? Is there a way to correlate finding that belong to other log type in a particular time range automatically? 

Reviewer: ```suggestion
But how would you know that the AD/LDAP detector finding `Invalid Username/Password, ResultType: 50126` is related to a chain of security events occurring in the same time frame on the infrastructure, such as a `403 Forbidden error` finding from an application logs detector? Is there a way to correlate finding that belong to other log types in a particular time range automatically? 
```

> But how would you know that the AD/LDAP detector finding `Invalid Username/Password, ResultType: 50126` is related to a chain of security events occurring in the same time window on the infrastructure, such a `403 Forbidden error` finding from an Application logs detector? Is there a way to correlate finding that belong to other log type in a particular time range automatically? 

Reviewer: If we answer these questions in the following paragraphs, we should say so here.

Author: Something like, "With correlation engine, you can." Or "Correlation engine makes all of the above possible." Or, "You can achieve all of these tasks through correlation engine."

Thoughts?

Reviewer: I like either the first or second one. If you use the second one, "this" instead of "the above".

> ## Benefits of correlation engine

Reviewer: ```suggestion
## Benefits of the correlation engine
```

> From inside OpenSearch Dashboards, the correlation engine provides a security finding knowledge graph that can be used to store connected findings and generate correlate insights based on the time windows from which those findings were logged. You can generalize the concept of “correlation” as “joining events across different data sources.”

Reviewer: ```suggestion
From inside OpenSearch Dashboards, the correlation engine provides a security finding knowledge graph that can be used to store connected findings and generate correlation insights based on the time ranges from which those findings were logged. You can generalize the concept of “correlation” as “joining events across different data sources.”
```

### Visualizing correlations

> To visualize correlations, log in to OpenSearch Dashboards and navigate to **Security Analytics > Overview > Correlations**. The correlations page shows a graph of findings connected to each other. The graph displays findings as nodes with colored borders expressing their severity level. A three-letter abbreviation inside the node indicates the log type. The lines that connect the findings represent the correlations between them. A heavy line indicates a strong correlation, while a light line shows a weaker connection, as shown in the following image.

Reviewer: ```suggestion
To visualize correlations, log in to OpenSearch Dashboards and navigate to **Security Analytics > Overview > Correlations**. The correlations page shows a graph of findings connected to each other. The graph displays findings as nodes with colored borders expressing their severity level. A three-letter abbreviation inside the node indicates the log type. The lines that connect the findings represent the correlations between them. A heavy line indicates a strong correlation, while a light line indicates a weaker correlation, as shown in the following image.
```

<img src="/assets/media/blog-images/2023-07-13-correlating-security-events/findings-graph.png" alt="A visualization of correlations"/>{: .img-fluid }

> When selecting a finding, a panel appears with detailed information about the finding including:

Reviewer: ```suggestion
When selecting a finding, a panel appears with detailed information about the finding, including the following:
```

* Log type
* Rule severity
* The time of the finding
* Rule information

> All findings also contain a **Correlation score**, a metrics that measures how strong the correlation is between two findings. The higher the correlation score, the stronger the correlation, as shown in the following image.

Reviewer: ```suggestion
All findings also contain a **correlation score**, a metric that measures how strong the correlation is between two findings. The higher the correlation score, the stronger the correlation, as shown in the following image.
```

<img src="/assets/media/blog-images/2023-07-13-correlating-security-events/correlation-score.png" alt="Correlation score"/>{: .img-fluid }

> You can file correlations by severity, log type, or by changing the time range.

Author: Sort might fit better.

> You can file correlations by severity, log type, or by changing the time range.

Author: ```suggestion
You can sort correlations by severity, log type, or by changing the time range.
```

## Creating a correlation rule

Correlation rules allow you to define threat scenarios involving multiple systems in an infrastructure by matching the signatures of threat events occurring in different log types. Once a rule contains at least two different log sources and the preferred fields and field values that define an intended threat scenario, the correlation engine can query the indexes specified in the correlation rule and identify any correlations between the findings.

For instructions on how to create correlation rule, see [Creating correlation rules](https://opensearch.org/docs/latest/security-analytics/sec-analytics-config/correlation-config/).

## Exploring findings

Finally, the **Findings** window includes features for viewing and working with findings. The two main features are:

* The bar graph with findings information arranged by count, date, and log type or rule severity.
* The **Findings** list arranged by time, finding ID, rule name, and other details.

For more information on using the Findings window, see [Working with findings](https://opensearch.org/docs/latest/security-analytics/usage/findings/).

> 

Author: Addressed this in line 118.

> 

Author: ```suggestion
## Next steps

To get started using correlation engine, see [Setting up Security Analytics](https://opensearch.org/docs/latest/security-analytics/sec-analytics-config/index/) and [Creating correlation rules](https://opensearch.org/docs/latest/security-analytics/sec-analytics-config/correlation-config/) in the OpenSearch documentation.
```

---
layout: post
title:  "Correlating security events across different log sources"
authors:
  - 
date: 2023-07-13
categories:
  - technical-posts
meta_keywords: OpenSearch security, OpenSearch security analytics, log analytics

> meta_description: This blog post details how Security Analytics can provide a better fidelity of data when viewed through a correlation graph, which can help security analysts monitor their systems and applications more effectively.

Reviewer: "*Meta_description*: Learn how the OpenSearch Security Analytics correlation engine helps you analyze different security logs and identify relationships between them to recognize correlated events."

For Security Analytics to monitor and detect threats for this infrastructure, we need to the define a detector for each component in the infrastructure. The following diagram shows the infrastructure with a detector for each component, such as a network detector for VPC Flow Logs and a Windows detector for the EC2 Windows instance.

<img src="/assets/media/blog-images/2023-07-13-correlating-security-events/detector-infrastructure.png" alt="Detector infrastructure inside of customer's application"/>{: .img-fluid }

With the detectors set, we can simulate a security attack on this example infrastructure. The attacker uses the `sbcd90` user to call a REST API named `POST /customer_records.txt`, which tries to replicate a sensitive file named `customer_records.txt` from Amazon S3.

When the attack occurs, each detector generates a finding for its corresponding infrastructure component. For example, the AD/LDAP detector generates a finding that indicates `Invalid Username/Password, ResultType: 50126`. This finding belongs to the individual detector's log type. 

> But how would you know that the AD/LDAP detector finding `Invalid Username/Password, ResultType: 50126` is related to a chain of security events occurring in the same time frame on the infrastructure, such as a `403 Forbidden error` finding from an application logs detector? Is there a way to correlate finding that belong to other log types in a particular time range automatically? 

Author: ```suggestion
But how would you know that the AD/LDAP detector finding `Invalid Username/Password, ResultType: 50126` is related to a chain of security events occurring in the same time frame on the infrastructure, such as a `403 Forbidden error` finding from an application logs detector? Is there a way to correlate finding that belong to other log types in a particular time range automatically? Correlation engine makes all of this possible.
```

## Creating a correlation rule

Correlation rules allow you to define threat scenarios involving multiple systems in an infrastructure by matching the signatures of threat events occurring in different log types. Once a rule contains at least two different log sources and the preferred fields and field values that define an intended threat scenario, the correlation engine can query the indexes specified in the correlation rule and identify any correlations between the findings.

For instructions on how to create correlation rules, see [Creating correlation rules](https://opensearch.org/docs/latest/security-analytics/sec-analytics-config/correlation-config/).

## Exploring findings

Finally, the **Findings** window includes features for viewing and working with findings. The two main features are:

* The bar graph with findings information arranged by count, date, and log type or rule severity.
* The **Findings** list arranged by time, finding ID, rule name, and other details.

For more information about using the Findings window, see [Working with findings](https://opensearch.org/docs/latest/security-analytics/usage/findings/).

## Next steps

> To get started using correlation engine, see [Setting up Security Analytics](https://opensearch.org/docs/latest/security-analytics/sec-analytics-config/index/) and [Creating correlation rules](https://opensearch.org/docs/latest/security-analytics/sec-analytics-config/correlation-config/) in the OpenSearch documentation.

Reviewer: ```suggestion
To get started using the correlation engine, see [Setting up Security Analytics](https://opensearch.org/docs/latest/security-analytics/sec-analytics-config/index/) and [Creating correlation rules](https://opensearch.org/docs/latest/security-analytics/sec-analytics-config/correlation-config/) in the OpenSearch documentation.
```
